ReportingServicesTools/Functions/Admin/Set-RsDatabase.ps1

```diff
@@ -157,37 +220,47 @@ function Set-RsDatabase
             Write-Verbose "Executing database creation script... Complete!"
         }
         #endregion Create Database if necessary
-        
+
         #region Configuring Database rights
-        # Step 3 - Generate database rights script
-        Write-Verbose "Generating database rights script..."
-        $isWindowsAccount = ($DatabaseCredentialType -like "Windows") -or ($DatabaseCredentialType -like "ServiceAccount")
-        $result = $rsWmiObject.GenerateDatabaseRightsScript($username, $Name, $IsRemoteDatabaseServer, $isWindowsAccount)
-        if ($result.HRESULT -ne 0)
-        {
-            Write-Verbose "Generating database rights script... Failed!"
-            throw "Failed to generate the database rights script from the report server using WMI. Errorcode: $($result.HRESULT)"
-        }
-        else
-        {
-            $SQLscript = $result.Script
-            Write-Verbose "Generating database rights script... Complete!"
-        }
-        
-        # Step 4 - Run Database rights script
-        Write-Verbose "Executing database rights script..."
-        try
+        if (-not $HasDatabaseUserRights)
```

Comment:
lets remove this condition and let the grant rights script run again it is idempotent and will simplify the testing and the code

Reply:
Updated.